junit-platform-suite-commons/src/main/java/org/junit/platform/suite/commons/SuiteLauncherDiscoveryRequestBuilder.java

```diff
@@ -198,6 +196,16 @@ public SuiteLauncherDiscoveryRequestBuilder suite(Class<?> suiteClass) {
 		return this;
 	}
 
+	public SuiteLauncherDiscoveryRequestBuilder suiteConfigurationParameters(Class<?> suiteClass) {
```

Comment:
I just realized we don't have any Javadoc for `SuiteLauncherDiscoveryRequestBuilder`. 😱 

So we should definitely address that, but rather in a different issue.

In any case, I'd like to see some Javadoc on this new method.

In addition, let's come up with a more descriptive name, perhaps something like one of the following.

- `configurationParametersForSuite(Class<?>)`
- `configurationParametersForSuiteClass(Class<?>)`
- `applyConfigurationParametersFromSuite(Class<?>)`

I know the last one is a bit long and doesn't start with `configurationParameter` like the existing methods; however, I _do_ think it aptly describes what it does (which is a bit special compared to the existing `configurationParameter()` and `configurationParameters()` methods).

Thoughts?

Reply:
I skipped the Javadoc because most classes marked as internal were lacking it. I'll add some for new methods.

Namingwise, I am considering deprecating the`suite` method and replace it with two methods:
 
 * `applyConfigurationParametersFromSuite`
 * `applySelectorsAndFiltersFromSuite`

And then also look at a way to avoid invoking `applyConfigurationParametersFromSuite` over and over again for evey unique id.

Reply:
> I skipped the Javadoc because most classes marked as internal were lacking it. I'll add some for new methods.

Indeed, it is marked as internal even though it's `public`. I had overlooked that.

In any case, I think it's still worth documenting (even if it's "just for us").

I raised #3695 to address that.

Speaking of which, would you like to address #3695?

I'm asking because someone else has already volunteered to do that, and I want to avoid duplicate work.

Reply:
> Namingwise, I am considering deprecating the existing `suite(Class<?>)` method and replacing it with two methods:
> 
>     * `applyConfigurationParametersFromSuite(Class<?>)`
> 
>     * `applySelectorsAndFiltersFromSuite(Class<?>)`

I like that. 👍 

I'm also wondering if we should add a precondition check which requires that the supplied `Class` is annotated with `@Suite`.

Or... perhaps that's over-engineering. 🤷 

> And then also look at a way to avoid invoking `applyConfigurationParametersFromSuite()` over and over again for every unique ID.

Yes, please do that.

> In hindsight these are two rather distinct operations. It just didn't occur to me when I wrote the suite engine.

No worries. It's a learning process, and APIs are meant to evolve. 😉

Reply:
> I'm asking because someone else has already volunteered to do that, and I want to avoid duplicate work.

That's quick! This looks like a good issue for a first time contributor, so I'm happy to see someone else try.

Reply:
> I'm also wondering if we should add a precondition check which requires that the supplied Class is annotated with @Suite.
>
> Or... perhaps that's over-engineering. 🤷

That would be a breaking change. The `@Suite` parameter is optional for the `JUnitPlatform` runner.